src/views/createArticle/index.jsx

```diff
@@ -0,0 +1,126 @@
+/* eslint-disable react/jsx-no-bind */
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Container, Alert } from 'react-bootstrap';
+import Create from '../../components/createForms/Create';
+import Preview from '../../components/previewArticle/Preview';
+import { validateArticle } from '../../utils/validator';
+import { createArticle } from '../../redux/actions/articles';
+import Header from '../../components/header/Header';
+import './createArticle.scss';
+
+function CreateArticle(props) {
+  const { user } = props;
+  const [articleData, setArticleData] = useState({
+    title: null,
+    description: null,
+    body: null,
+    image: null
+  });
+  const [state, setState] = useState({
+    component: 'write',
+    loading: false,
+    errors: [],
+    successMessage: ''
+  });
+
+  const {
+    component,
+    loading,
+    errors,
+    successMessage
+  } = state;
+
+  async function completeArticleCreation() {
+    setState({ ...state, errors: [], loading: true });
+    const res = await props.createArticle(articleData);
+    const { errors: errorsArr } = res;
+    setState({ ...state, loading: false });
+    if (errorsArr) {
+      setState({ ...state, errors: errorsArr });
+    }
+    setState({ ...state, successMessage: res.message });
+    props.history.push(`/articles/${res.article.slug}`);
+  }
+
+  function runValidators(e) {
+    e.preventDefault();
+    const validationErrors = validateArticle(articleData);
+    if (validationErrors.length > 0) {
+      return setState({ ...state, component: 'write', errors: validationErrors });
+    }
+    return completeArticleCreation();
+  }
+
+  function updateInput(e) {
```

Comment:
@leksyib please use `event` instead of `e`

Reply:
Thanks will do.